src/TrackGenerator.cpp

```diff
@@ -268,9 +273,27 @@ double TrackGenerator::getZCoord() {
 
 
 /**
+ * @brief Return the array used to store the FSR volumes
+ * @return _FSR_volumes the FSR volumes array indexed by FSR ID
+ */
+FP_PRECISION* TrackGenerator::getFSRVolumesBuffer() {
```

Comment:
Is it true that this is the only place in the code where we have a special routine to allocate a memory buffer? For better or worse, every array that we allocate is done inline within the routine that initializes the array. For consistency I would say simply do the same here, but can be convinced otherwise if there is a good reason for this allocation to be in its own method.